### PR TITLE
feat: raise exception when docker version is below 20.0.0

### PR DESCRIPTION
**Pull request by alaeddine-13**

### CODE REVIEW

File: jina/peapods/runtimes/container/__init__.py [hunk 1]
```diff
@@ -81,6 +81,16 @@ def _docker_run(self, replay: bool = False):
 
         client = docker.from_env()
 
+        docker_version = client.version().get("Version")
```
Comment (JoanFM) [inline jina/peapods/runtimes/container/__init__.py:84]: please use `single quotes`

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by JoanFM** [comment]
Why is this then not shown in the `extra-requirements.txt`?

**Comment by alaeddine-13** [comment]
> Why is this feature required?

When I was trying to run an executor with jinahub, I receive
```
docker.errors.APIError: 400 Client Error for http+docker://localhost/v1.40/containers/create?name=pod0ContainerRuntime: Bad Request ("invalid IP address in add-host: "host-gateway"")
```
Upgrading docker's version from 19 to 20 solves the issue

**Comment by JoanFM** [comment]
> > Why is this feature required?
> 
> When I was trying to run an executor with jinahub, I receive
> 
> ```
> docker.errors.APIError: 400 Client Error for http+docker://localhost/v1.40/containers/create?name=pod0ContainerRuntime: Bad Request ("invalid IP address in add-host: "host-gateway"")
> ```
> 
> Upgrading docker's version from 19 to 20 solves the issue

I think we should understand what's wrong and what exact version is solving this. I think the reason that bumping version solved this is not so strong.

**Comment by alaeddine-13** [comment]
> > > Why is this feature required?
> > 
> > 
> > When I was trying to run an executor with jinahub, I receive
> > ```
> > docker.errors.APIError: 400 Client Error for http+docker://localhost/v1.40/containers/create?name=pod0ContainerRuntime: Bad Request ("invalid IP address in add-host: "host-gateway"")
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Upgrading docker's version from 19 to 20 solves the issue
> 
> I think we should understand what's wrong and what exact version is solving this. I think the reason that bumping version solved this is not so strong.

I found this issue where they discuss the problem
https://github.com/docker/cli/issues/2664
According to it, the 20.0x versions of the docker daemon support `host.docker.internal:host-gateway`

**Comment by JoanFM** [comment]
> > > > Why is this feature required?
> > > 
> > > 
> > > When I was trying to run an executor with jinahub, I receive
> > > ```
> > > docker.errors.APIError: 400 Client Error for http+docker://localhost/v1.40/containers/create?name=pod0ContainerRuntime: Bad Request ("invalid IP address in add-host: "host-gateway"")
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Upgrading docker's version from 19 to 20 solves the issue
> > 
> > 
> > I think we should understand what's wrong and what exact version is solving this. I think the reason that bumping version solved this is not so strong.
> 
> I found this issue where they discuss the problem
> [docker/cli#2664](https://github.com/docker/cli/issues/2664)
> According to it, the 20.0x versions of the docker daemon support `host.docker.internal:host-gateway`

Good finding then! Let's do this!

**Comment by JoanFM** [comment]
> > > > Why is this feature required?
> > > 
> > > 
> > > When I was trying to run an executor with jinahub, I receive
> > > ```
> > > docker.errors.APIError: 400 Client Error for http+docker://localhost/v1.40/containers/create?name=pod0ContainerRuntime: Bad Request ("invalid IP address in add-host: "host-gateway"")
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Upgrading docker's version from 19 to 20 solves the issue
> > 
> > 
> > I think we should understand what's wrong and what exact version is solving this. I think the reason that bumping version solved this is not so strong.
> 
> I found this issue where they discuss the problem
> [docker/cli#2664](https://github.com/docker/cli/issues/2664)
> According to it, the 20.0x versions of the docker daemon support `host.docker.internal:host-gateway`

May u reference this issue in the code or even in the `raised exception`?

**Review by JoanFM — APPROVED** [review]

**Pull request merged**